web/src/tools/building/sourcemap-root/index.ts

```diff
@@ -75,16 +75,9 @@ for(let procArgIndex = 4; procArgIndex < process.argv.length; procArgIndex++) {
 let srcMap = SourcemapRemapper.fromFile(destFile);
 
 if(shouldClean) {
-  // First pass:  remove any closure prepended ../../../ pathing.
-  srcMap.remapPaths([
-    { from: "../../../", to: "" }
-  ]);
-
-  // Second pass:  fix up any mangled source paths
-  srcMap.remapPaths([
-    { from: "common/web/input-processor/build/keyman/", to: "" },
-    { from: "common/web/keyboard-processor/build/keyman/", to: "" }
-  ]);
+  // Keeps "@keymanapp/keyman" as sourceRoot, but prepends all `sources` entries with
+  // the remainder / suffix of sourceRoot and normalizes the paths.
+  srcMap.normalizeWithSourceRoot('@keymanapp/keyman/'.length);
```

Comment:
For clarity, before the normalization pass, the sourceRoot for app/browser is set here, to force proper repo layout on the sources post-esbuild:

https://github.com/keymanapp/keyman/blob/c594186f958f6c6b82dc16a4f7069538ea9436fb/web/src/app/browser/build-bundler.js#L38-L41

As our affected sourcemaps all have this as their `sourceRoot` prefix, passing the index of the character following the prefix here tells the `normalizeWithSourceRoot` function to split the existing sourceRoot there and prepend all `sources` entries with all trailing components.  These differ per target:  for this example, `web/build/app/browser/debug`.  Post-normalization, that gives us a perfect match to the repo's layout.